### PR TITLE
feat: add transformMessage method

### PR DESCRIPTION
**Pull request by Garma00**
#### List of Changes

* Add a method that transform the result of the `extractMessage` method into a valid type for the `loadMessage` method;

#### Motivation and Context

We need to transform the output before loading it.

#### How Has This Been Tested?

Unit test.

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] My changes does not requires change to the documentation,

### CODE REVIEW

File: apps/etl-func/src/adapters/message.ts [hunk 1]
```diff
@@ -38,4 +40,42 @@ export class MessageAdapter implements MessageRepository {
       throw error;
     }
   }
+
+  /**
+   * Transform a Message into a MessageEvent
+   **/
+
+  getMessageEventFromMessage({
```
Comment (lucacavallaro) [inline apps/etl-func/src/adapters/message.ts]: imho this should be a **pure** function unrelated to `Message` class

`Message` should not be aware of `MessageEvent` 😄

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: apps/etl-func/src/adapters/message.ts (hunk 1 above)
Comment (Garma00) [reply apps/etl-func/src/adapters/message.ts]: Yess you are right, take a look here [5529dcf](https://github.com/pagopa/io-messages/pull/58/commits/5529dcf7a32c8ed41aac111efe5569703a63f4c0) 😄

File: apps/etl-func/src/domain/message.ts [hunk 2]
```diff
@@ -13,13 +13,18 @@ export const messageMetadataSchema = z.object({
   fiscalCode: z.string().min(1),
   id: z.string().ulid(),
   indexedId: z.string().ulid(),
-  isPending: z.boolean().optional(),
+  isPending: z.boolean(),
```
Comment (Garma00) [inline apps/etl-func/src/domain/message.ts:16]: @lucacavallaro as we discussed I removed the `optional` flag from isPending.
The function that calls `getMessageByMetadata` method will be responsible for ensuring that `isPending` is defined.

File: apps/etl-func/src/adapters/message.ts [hunk 3]
```diff
@@ -1,13 +1,48 @@
...
+/**
+ * Transform a Message into a MessageEvent
+ **/
+
+export const getMessageEventFromMessage = ({
+  content,
+  contentType,
+  id,
+  metadata,
+}: Message): MessageEvent => ({
```
Comment (lucacavallaro) [inline apps/etl-func/src/adapters/message.ts]: add `messageEventSchema.parse`

File: apps/etl-func/src/adapters/message.ts [hunk 4]
```diff
@@ -1,13 +1,48 @@
...
+  is_pending: metadata.isPending ?? false,
+  op: "CREATE",
+  payment_data_amount: content.payment_data?.amount ?? null,
+  payment_data_invalid_after_due_date:
+    content.payment_data?.invalid_after_due_date ?? null,
+  payment_data_notice_number: content.payment_data?.notice_number ?? null,
+  payment_data_payee_fiscal_code:
+    content.payment_data?.payee?.fiscal_code ?? null,
+  require_secure_channels: content.require_secure_channels,
+  schema_version: 1,
```
Comment (lucacavallaro) [inline apps/etl-func/src/adapters/message.ts]: `.default(1)`

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

**Review by lucacavallaro — APPROVED** [review]

**Pull request merged**